sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -39,22 +38,43 @@ private[csv] object CSVInferSchema {
    *     3. Replace any null types with string type
    */
   def infer(
```

Comment:
This is kind of a important change to introduce similar functionalities with JSON. (e,g., creating a dataframe from `RDD[String]` or `Dataset[String]`).